subsys/net/lib/download_client/src/download_client.c

```
@@ -197,6 +222,10 @@ static int client_connect(struct download_client *dl, const char *host,
 		}
 	}
 
+	if (dl->config.sec_tag == -1 && dl->config.set_tls_hostname) {
+		LOG_WRN("set_tls_hostname flag is set for non-TLS connection");
+	}
```

We should return an error here, perhaps `EINVAL`